src/App.js

```diff
@@ -1,86 +1,201 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { TodoList } from './components/TodoList';
+import { Filters } from './components/Filters';
+
+const useLocalStorage = (key, initialValue) => {
+  const [value, setValue] = useState(
+    JSON.parse(localStorage.getItem(key)) || initialValue,
+  );
+  const save = (newValue) => {
+    setValue(newValue);
+    localStorage.setItem(key, JSON.stringify(newValue));
+  };
+
+  return [value, save];
+};
 
 function App() {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useLocalStorage('todos', []);
+  const [filteredTodos, setfilteredTodos] = useState(todos);
+  const [filterStatus, setFilterStatus] = useState('');
+  const [activeSelectAll, setActiveSelectAll] = useState(false);
+
+  const addTodo = (value) => {
+    setTitle(value);
+  };
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+
+    if (title.length === 0) {
+      return;
+    }
+
+    const newTodo = {
+      title,
+      id: uuidv4(),
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
+    setfilteredTodos([...todos, newTodo]);
+
+    setTitle('');
+  };
+
+  const toggleAll = () => {
+    const uncheckedTodos = todos.map(todo => (
+      {
+        ...todo,
+        completed: false,
+      }
+    ));
+
+    setTodos(uncheckedTodos);
+    setfilteredTodos(uncheckedTodos);
+  };
+
+  const changeStatus = (checkedTodoId) => {
+    const todosCopy = todos.map((todo) => {
+      if (todo.id === checkedTodoId) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+
+      return todo;
+    });
+
+    setTodos(todosCopy);
+    setfilteredTodos(todosCopy);
```

Comment:
no need every time update both todos and filtered todos, also no need to put filtered todos in state, let's create a variable filtered todos, which will be updated every time the filterStatus changed